uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/NodeMouseEventHandler.java

```diff
@@ -81,4 +83,28 @@ default boolean handleBodyCell(final GridWidget gridWidget,
                                    final AbstractNodeMouseEvent event) {
         return false;
     }
+
+    /**
+     * Returns whether the {@link AbstractNodeMouseEvent} occurred during a Drag and Drop operation.
+     * @param gridWidget The {@link GridWidget} on which the event occurred.
+     * @return true if the event occurred during a Drag and Drop operation.
+     */
+    default boolean isDNDOperationInProgress(final GridWidget gridWidget) {
+        if (!(gridWidget.getLayer() instanceof GridLayer)) {
+            return false;
+        }
+        final GridLayer gridLayer = (GridLayer) gridWidget.getLayer();
+        final GridWidgetDnDHandlersState.GridWidgetHandlersOperation operation = gridLayer.getGridWidgetHandlersState().getOperation();
+        switch (operation) {
+            case NONE:
+            case COLUMN_RESIZE_PENDING:
+            case COLUMN_MOVE_PENDING:
+            case COLUMN_MOVE_INITIATED:
+            case ROW_MOVE_PENDING:
+            case ROW_MOVE_INITIATED:
+            case GRID_MOVE_PENDING:
```

Comment:
Reading the javadoc. What we mean by Drag and Drop? My understanding was button pressed and moving mouse what would mean here we should have just constants with MOVE in name, not PENDING or INITIATED. So is Drag and Drop also not moving mouse but keeping the mouse button pressed?

Reply:
Hi @jomarko a DnD operation is indeed any mousedown, mousemove, mouseup sequence of events. Furthermore a click event is a combination of mousedown, mouseup on the same HTML element (with or without a mousemove in between). Browsers detect the mousedown, mouseup and raise a separate click event in addition to the mousedown and mouseup events.

`_PENDING` means the pointer is _over_ a header cell (or "resize hot spot" etc) but a mousedown event has not happened (i.e. User is simply moving the pointer over a cell). `_INITIATED` means a previous `_PENDING` operation was detected and now a mousedown event has occurred and the DnD operation has been _initiated_ (but may result in nothing happening, if the User does not actually move the column). When the column has actually been moved the DnD state is moved to `_MOVE`.

Therefore, when single left clicking on a header (to edit it), mousedown, mouseup and click events are raised by the browser. This led to the "edit header" popup showing after _moving_ a column to a new position which was undesirable. To prevent this from happening I added the `COLUMN_MOVE_INITIATED` state that means the cursor was over a movable column and a mousedown event was received BUT the column has NOT been moved to a new position.

In the `isDNDOperationInProgress(..)` [code](https://github.com/kiegroup/appformer/pull/893/files#diff-c4a591df74dd075a2c1ae794a9aa2bb2R102) `COLUMN_MOVE_INITIATED` does not signal a DnD operation has actually happened (i.e. the column has not moved) and therefore both `DefaultGridWidgetCellSelectorMouseEventHandler` or `DefaultGridWidgetEditCellMouseEventHandler` respond to the click event to which they're bound. This is means when a user that a clicks on a header cell it is both selected and its editor is launched.

However IF a column has actually been moved then a DnD operation has actually happened and both of the foregoing handlers do nothing, meaning the column is moved but it is neither selected nor its editor launched.

I hope this explains the change (slightly!)